unittests/savanna_cluster.hpp

```diff
@@ -230,51 +262,91 @@ namespace savanna_cluster {
 
       void reset_lib() { for (auto& n : _nodes) n.reset_lib();  }
 
+      void push_block(size_t dst_idx, const signed_block_ptr& sb) {
+         push_block_to_peers(dst_idx, false, sb);
+      }
+
+      void verify_lib_advances() {
+         auto lib = _nodes[0].lib_block->block_num();
+         size_t tries = 0;
+         while (_nodes[0].lib_block->block_num() <= lib + 3 && ++tries < 10) {
```

Comment:
The `tries` should not be needed.

Reply:
I think they are needed if we want to see the "BOOST_REQUIRE` error message instead of the test not terminating.

Reply:
Yes, we do. This is a convenience function that we can just throw at the end of a test to make sure we are back to a state where lib advances when blocks are produced.